lib/services/api.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:adventures_in_chat_app/models/message.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/foundation.dart';
+
+typedef CollectionGet = CollectionReference Function(String path);
+
+class Api {
+  final CollectionGet collectionGet;
```

Comment:
I found passing in a function to be stored as a member to be a bit of a head scratcher but I'm probably missing something. Why not inject a Firestore object?  Then the service could do more as it's not limited to collection calls.

Reply:
I guess I wrote it like this because we're not currently using any other method of Firestore, but it makes more sense to pass the Firestore instance instead. Fixing it now.